src/core_modules/capture-core/components/Pages/Search/SearchPage.actions.js

```diff
@@ -1,48 +1,21 @@
+import { createAction } from '@reduxjs/toolkit';
 import { actionCreator } from '../../../actions/actions.utils';
 
 export const searchPageActionTypes = {
-    VIA_UNIQUE_ID_ON_SCOPE_PROGRAM_SEARCH: 'SearchViaUniqueIdOnScopeProgram',
-    VIA_UNIQUE_ID_ON_SCOPE_TRACKED_ENTITY_TYPE_SEARCH: 'SearchViaUniqueIdOnScopeTrackedEntityType',
-    VIA_ATTRIBUTES_ON_SCOPE_PROGRAM_SEARCH: 'SearchViaAttributesOnScopeProgram',
-    VIA_ATTRIBUTES_ON_SCOPE_TRACKED_ENTITY_TYPE_SEARCH: 'SearchViaAttributesOnScopeTrackedEntityType',
-    SEARCH_RESULTS_LOADING_VIEW: 'SearchResultsLoadingView',
-    SEARCH_RESULTS_EMPTY_VIEW: 'SearchResultsEmptyView',
-    SEARCH_RESULTS_SUCCESS_VIEW: 'SearchResultsSuccessView',
-    SEARCH_RESULTS_ERROR_VIEW: 'SearchResultsErrorView',
-    SEARCH_RESULTS_INITIAL_VIEW: 'SearchResultsInitialView',
     TO_MAIN_PAGE_NAVIGATE: 'NavigateToMainPage',
-    CURRENT_SEARCH_INFO_SAVE: 'SaveCurrentSearchInfo',
-    INITIAL_PROGRAM_ID_STORE: 'StoreInitialProgramId',
 };
 
-export const searchViaUniqueIdOnScopeTrackedEntityType = ({ trackedEntityTypeId, formId }) =>
-    actionCreator(searchPageActionTypes.VIA_UNIQUE_ID_ON_SCOPE_TRACKED_ENTITY_TYPE_SEARCH)({ trackedEntityTypeId, formId });
+export const searchViaUniqueIdOnScopeTrackedEntityType = createAction('search/searchViaUniqueIdOnScopeTrackedEntityType');
+export const searchViaUniqueIdOnScopeProgram = createAction('search/searchViaUniqueIdOnScopeProgram');
+export const searchViaAttributesOnScopeTrackedEntityType = createAction('search/searchViaAttributesOnScopeTrackedEntityType');
+export const searchViaAttributesOnScopeProgram = createAction('search/searchViaAttributesOnScopeProgram');
 
-export const searchViaUniqueIdOnScopeProgram = ({ programId, formId }) =>
-    actionCreator(searchPageActionTypes.VIA_UNIQUE_ID_ON_SCOPE_PROGRAM_SEARCH)({ programId, formId });
+export const showInitialViewOnSearchPage = createAction('search/showInitialViewOnSearchPage');
+export const showLoadingViewOnSearchPage = createAction('search/showLoadingViewOnSearchPage');
+export const showSuccessResultsViewOnSearchPage = createAction('search/showSuccessResultsViewOnSearchPage');
+export const showEmptyResultsViewOnSearchPage = createAction('search/showEmptyResultsViewOnSearchPage');
+export const showErrorViewOnSearchPage = createAction('search/showErrorViewOnSearchPage');
+export const saveCurrentSearchInfo = createAction('search/saveCurrentSearchInfo');
```

Comment:
So basically now we dont have to write strings as types. To refer the type of an action you simply refer the searchViaUniqueIdOnScopeProgram which is the result of the createAction redux-toolkit provides.

This has some advantages to what plain the redux solution provides. As for instance boilderplate.